Src/FluentAssertions/Primitives/StringContainsStrategy.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Primitives;
+
+internal class StringContainsStrategy : IStringComparisonStrategy
+{
+    private readonly IEqualityComparer<string> comparer;
+    private readonly OccurrenceConstraint occurrenceConstraint;
+    private readonly bool ignoringCase;
+
+    public StringContainsStrategy(IEqualityComparer<string> comparer, OccurrenceConstraint occurrenceConstraint)
+    {
+        this.comparer = comparer;
+        this.occurrenceConstraint = occurrenceConstraint;
+        ignoringCase = comparer.Equals("A", "a");
```

Comment:
🤔 I get a strong feeling we should not take an `IEqualityComparer<string>`, but just use a `StringComparer`.

Reply:
What would be the advantage, when I don't use any other property or method than the one from the interface?

Reply:
1. It would avoid the type check in `Using`
2. It would avoid the weird logic to determine whether compare is case insensitive.

Reply:
> 1. It would avoid the type check in `Using`

Yes, I could do this, but I could do this as well with an explicit additional overload for `IEqualityComparer<string>`. Would this really be better?

> 2. It would avoid the weird logic to determine whether compare is case insensitive.
How so? Also a `StringComparer` does not tell me, if it is case-sensitive or not? I could always create a custom `StringComparer` and would then need to check in the strategy how it handles case-sensitivity.

Reply:
> How so? Also a `StringComparer` does not tell me, if it is case-sensitive or not? I could always create a custom `StringComparer` and would then need to check in the strategy how it handles case-sensitivity.

Than you can check which `StringComparer` it is, e.g. `StringComparer.StringContainsStrategy`

Reply:
I can still write custom `StringComparer` and would have the same issue with it. Additionally it would no longer solve the issue in #2339 (or would require the user to write a custom `StringComparer` with two additional methods).

I only need this information (if it is case sensitive or not) for the `ExpectationDescription`, e.g. in `StringContainsStrategy`:
```csharp
    string predicateDescription = ignoringCase ? "contain the equivalent of" : "contain";
    return "Expected {context:string} to " + predicateDescription + " ";
```

What if I always print out "contain the equivalent of" when calling `Should().ContainEquivalentOf`?

Reply:
Ah yes, you're right. Ignore this comment. However, we do need to add an overload of `SelfReferenceEquivalencyOptions<TSelf>.Using`

Reply:
I got rid of the strange `comparer.Equals("A", "a")` in [42bb1d2c](https://github.com/fluentassertions/fluentassertions/pull/2413/commits/42bb1d2c766d024c81d46341dea9036c1cfffc2a).

Reply:
We try to avoid boolean parameters. And since it feels a bit weird how that boolean is used inside the strategies, why not replace the boolean with the actual string that it results in?

Reply:
Great idea. I changed it accordingly!